src/app/itemsOnLifecycle.clj

```diff
@@ -161,4 +161,26 @@
 ; the param :onlyIncorrect
 ; Either way it produces a table with two columns: class and problem. Problem can be potentially empty.
 (def utilsClassesOp
-  (Operation. utilsClassesQuery [:onlyIncorrect] [:class :problem]))
\ No newline at end of file
+  (Operation. utilsClassesQuery [:onlyIncorrect] [:class :problem]))
+
+; ============================================
+; ITEM 7
+; ============================================
+
+(defn calls-finalizers? [class]
+  (->> (getMethods class)
+       (map toStringWithoutComments)
+       (filter #(.contains % "finalize("))
```

Comment:
This would work most of the times but it could lead to problems if the for example the line is commented or this calls a method called finalize taking more than zero params.

In my opinion the ideal way to do that is looking for nodes of type MethodCallExpr (see https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/ast/expr/MethodCallExpr.java) with name "finalize" and passing zero params.

To find all nodes in a Class we could either call getChildrenNodes recursively (see https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/ast/Node.java) or we could extend TreeVisitor (see https://github.com/javaparser/javaparser/blob/b2b0d7b20395404061b10b774200f922c0b7b6c3/javaparser-core/src/main/java/com/github/javaparser/ast/TreeVisitor.java). 

Probably we need methods to get all the nodes and maybe also all the nodes of a certain kind. 
If you want to use this approach I would then make it more general maybe, otherwise if you prefer I can first add the general functions and then you can use them for this case.